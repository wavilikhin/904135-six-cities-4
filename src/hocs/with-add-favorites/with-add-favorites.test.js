import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import NameSpace from '../../reducer/name-space.js';
import withAddFavorites from './with-add-favorites.jsx';

const MockComponent = () => {
  return <div></div>;
};
const mockStore = configureStore([]);

const MockComponentWrapped = withAddFavorites(MockComponent);

const userFavorites = [];
const offersArray = [];
const handleFavoritesChange = () => {};

describe(`WithAddFavorites component snapshot test`, () => {
  it(`Should render wrapped component well`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        offers: [],
      },
      [NameSpace.STATE]: {
        city: '',
      },
      [NameSpace.USER]: {
        authStatus: 'AUTH',
        userEmail: '',
        userFavorites: [],
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <MockComponentWrapped
            userFavorites={userFavorites}
            offersArray={offersArray}
            handleFavoritesChange={handleFavoritesChange}
          />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
