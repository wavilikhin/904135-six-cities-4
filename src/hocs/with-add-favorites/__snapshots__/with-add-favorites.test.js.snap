// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithAddFavorites component snapshot test Should render wrapped component well 1`] = `<div />`;
